service/chartconfig/v1/resource/chart/create.go

```diff
@@ -29,6 +30,12 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 		if err != nil {
 			return microerror.Mask(err)
 		}
+		defer func() {
+			err := os.Remove(tarballPath)
```

Comment:
The clients are configured with an FS. We should have the same kind of abstraction here to be set for the future. Also makes it more testable.

Reply:
Ok, i'll add the dependency on a separate PR.